packages/desktop/src/renderer/index.tsx

```diff
@@ -30,7 +27,20 @@ const persistentConfigService = new PersistentConfigServiceElectron();
 const applicationInfoService = new ApplicationInfoServiceElectron();
 const databaseService = new DatabaseServiceElectron();
 const executionEnvService = new ExecutionEnvServiceElectron(notificationService);
-// application analytics/metrics
+
+// Define the KeyDownHandler component inline
+const KeyDownHandler: React.FC<{ clearStore: () => void }> = ({ clearStore }) => {
```

Comment:
I think creating a component just to wrap a side-effect is more work than necessary. In `renderFmsFileExplorer` you should be able to add the `useKeyDown` hook call right before the `render`

Reply:
Does this still apply now that we are reloading within the component?

Reply:
I think so? `window.location.reload()` is such a major side effect (wiping the whole DOM) the behavior is probably the same regardless of whether it's in a component?

Reply:
Yes, this is still relevant. Essentially it is bad style to create a component for the purpose of a side-effect. Components are for rendering elements.

Reply:
Update on this. Hooks have to be called within a component according to this error:

```
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (react.development.js:1476)
    at Object.useEffect (react.development.js:1519)
    at useKeyDown (index.ts:4)
    at renderFmsFileExplorer (index.tsx:108)
    at Module../src/renderer/index.tsx (index.tsx:130)
    at __webpack_require__ (bootstrap:24)
    at startup:6
    at startup:6
```